### DISCUSSION TITLE
How to set "SINGULARITY_SHELL" variable?

### DISCUSSION THREAD

**Discussion by Heng-Zhou** [sylabs/singularity#3363] [Q&A] [answered]
I wish to set `SINGULARITY_SHELL` variable which is used in `/.singularity.d/actions/shell` to control what shell to use and how to start it. I tried to set it in `%environment` and `%post`, but none of them work. So, how to set this variable?

**Comment by Heng-Zhou** [comment]
It is set in %environment section. I did not spot it out previously.

**Comment by dtrudg** [comment, marked as answer]
Please see the `singularity shell --help` output and the environment variable documentation. The shell is set via the `-s` or `--shell` flag:

```
  -s, --shell string                  path to program to use for
                                      interactive shell
```

Or by setting `SINGULARITY_SHELL` *on the host*.

Manipulating the variable inside a definition file / container directly is not something that we advise. The exact code of the action scripts is considered to be an internal detail, and the way that the action scripts are implemented in newly built containers may change in future.

The intended / supported behaviour is only by setting `SINGULARITY_SHELL` on the host, or use the `-s / --shell` flag.
